aten/src/ATen/core/TensorImpl.h

```diff
@@ -1405,6 +1405,8 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
 
 };
 
+#if !defined _MSC_VER || !defined(_DEBUG)
```

Comment:
Why do we need this change?

Reply:
For some unknown for me reason in case of Debug build in Visual Studio 2015
sizeof(TensorImpl) == sizeof(int64_t) * 13
instead of 
sizeof(TensorImpl) == sizeof(int64_t) * 12
which leads to the failure of static_assert.
It can be unused variable(s) that is optimized during release build or just some additional compiler stuff used for debugging purposes. I didn't investigate the real reason. 
This is added just to make debug build with Visual Studio 2015 possible.

Reply:
Added a comment on that for clarification.

Reply:
@peterjc123 @orionr 
This one is somehow fixed on its own after last night master merge. At least here some good news :)
Removed `#if !defined _MSC_VER || !defined(_DEBUG)`